### PR TITLE
[qt5-base] Add zstd as a dependency

### PR DESCRIPTION
**Pull request by Neumann-A**
also moves embree3 dependency to the correct port.
also moves the hdf5 fixes to vtk via applying an upstream patch. 

For testing: Added `Default-Features: vtkm, tools` to paraview.

### CODE REVIEW

File: ports/paraview/portfile.cmake [hunk 1]
```diff
@@ -65,13 +65,9 @@ if("python" IN_LIST FEATURES)
         -DPython3_FIND_REGISTRY=NEVER
         "-DPython3_EXECUTABLE:PATH=${PYTHON3}" # Required by more than one feature
         )
-  
     #VTK_PYTHON_SITE_PACKAGES_SUFFIX should be set to the install dir of the site-packages
 endif()
```
Comment (Neumann-A) [inline ports/paraview/portfile.cmake:70]:
```suggestion
string(APPEND VCPKG_C_FLAGS " -DH5_USE_110_API")
string(APPEND VCPKG_CXX_FLAGS " -DH5_USE_110_API")
```

**Comment by JackBoosY** [comment]
Please ping me if this PR ready for review.

**Comment by Neumann-A** [comment]
@JackBoosY: Can I keep feature `tools` in paraview as a default-feature? It is basically testing if Qt and VTK are linking everything correctly. If CI is green this should be good to go.

**Comment by JackBoosY** [comment]
@Neumann-A That's okay.

**Comment by Neumann-A** [comment]
@JackBoosY: ready

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
I think a better approach than enabling the tools by default would be to add a new helper port (`vcpkg-ci-test-helper`) which depends on the features we want built in CI. That way, need for CI testing can be separated from what users actually need by default.

**Comment by Neumann-A** [comment]
@ras0219-msft: The problem here is I cannot deactivate zstd without patching qt's configure.json somewhere. zstd will be picked up if it is available either within vcpkg or by the system. So it is better to activate it so that the vcpkg built version always gets picked up.

**Comment by Neumann-A** [comment]
@ras0219-msft: disregard you probably meant the paraview tools change ;)

**Comment by Neumann-A** [comment]
@ras0219-msft: maybe vcpkg should use a port overlay for CI for ports testing other ports in CI? So the main port dir does not get cluttered. maybe something like `ci-ports` or `ports-ci` or `ports-testing`

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil** [comment]
@Neumann-A Could you merge with master?

**Comment by strega-nil** [comment]
I'm gonna wait until after #12529 and rerun the CI.

**Review by strega-nil — APPROVED** [review]

**Pull request merged**